libbeat/outputs/elasticsearch/client.go

```diff
@@ -341,6 +341,23 @@ func isRequireAliasSupported(version common.Version) bool {
 	return !version.LessThan(common.MustNewVersion("7.10.0"))
 }
 
+// isAlias determines if the target is an alias or an index.
+// If meta.alias is set in the event, it is an alias because the
+// field takes precedence the over the contents of index and raw_index.
+// If nothing is set, it is deemed to be an alias.
+func isAlias(event *beat.Event) bool {
+	if v, _ := events.GetMetaStringValue(*event, events.FieldMetaAlias); v != "" {
+		return true
+	}
+	if v, _ := events.GetMetaStringValue(*event, events.FieldMetaIndex); v != "" {
```

Comment:
I wonder what all the places are where we us this. I thought we also use this in places like k8s do adjust / overwrite the automatically created index name. In these scenarios this would not be correct.

An alternative approach is to introduce `events.MetaInternalIndex` or similar that is purely set by the stack monitoring modules so it should not have any side effects.

One more idea, we hardcode that if the index name matches `.monitoring-*` we don't do the check.

I'm worried that if we are not very specific with the check, it will have unexpected side effects.

Reply:
> I wonder what all the places are where we us this. I thought we also use this in places like k8s do adjust / overwrite the automatically created index name. In these scenarios this would not be correct.

I am not sure I understand why it would be an issue. If anything overwrites the `index` or `raw_index`, we do not set `require_alias`, so there is no problem during ingestion. If something sets `alias`, the target is an alias so `require_alias` should be safe.

> One more idea, we hardcode that if the index name matches .monitoring-* we don't do the check.

I also considered it, but I discarded the idea because I do not want to write monitoring specific code in a general elasticsearch client code.

Reply:
> I am not sure I understand why it would be an issue. If anything overwrites the index or raw_index, we do not set require_alias, so there is no problem during ingestion. If something sets alias, the target is an alias so require_alias should be safe.

The initial change was introduced to fix an issue when an index is deleted. Now lets assume for a moment (alias partially made up) the metricset `kubernetes.foo_bar` sets the alias name to `kubernetes.foo.bar` specifically. This means the alias feature is disabled. Now the user wipes the alias with all indices and we start to ingest data into `kuberentes.foo.bar`. If we see a support request for this, our first answer is this is not possible. But turns out, it is in this case. What you have will fix most cases but it creates incosistency. Lets rather revert the previous change and discuss on how we should deal with it.